src/dataframe/dataframe.jl

```diff
@@ -994,6 +994,8 @@ Base.convert(::Type{DataFrame}, d::AbstractDict) = DataFrame(d)
 ##
 ##############################################################################
 
+Base.push!(df::DataFrame, r::DataFrameRow) = append!(df, parent(r)[row(r),:])
```

Comment:
Unfortunately this implementation isn't really efficient, which can be a problem if you repeatedly push rows in a loop. I think we'd better allow iterating over a `DataFrameRow`, which is consistent with `NamedTuple`. Then the already existing method below should work automatically.

Reply:
`DataFrameRow`s are already iterable but are incompatible with the `push!` for iterable defined below:
```julia
julia> df = DataFrame(A = 1:2, B = 'a':'b')
2×2 DataFrames.DataFrame
│ Row │ A │ B   │
├─────┼───┼─────┤
│ 1   │ 1 │ 'a' │
│ 2   │ 2 │ 'b' │

julia> r = DataFrameRow(df, 2)
DataFrameRow (row 2)
A  2
B  b


julia> collect(r)
2-element Array{Tuple{Symbol,Any},1}:
 (:A, 2)  
 (:B, 'b')

julia> push!(df, collect(r))
ERROR: ArgumentError: Error adding (:A, 2) to column :A. Possible type mis-match.
Stacktrace:
 [1] push!(::DataFrames.DataFrame, ::Array{Tuple{Symbol,Any},1}) at /home/omus/.julia/v0.6/DataFrames/src/dataframe/dataframe.jl:1034
```

Reply:
Ah, right. I think we should change this, so that `DataFrameRow` and `NamedTuple` can be used interchangeably (and the former is just a mutable version of the latter). It will be breaking, but hopefully not many people depend on it.

Reply:
Actually we cannot change it without a deprecation period (https://github.com/JuliaData/DataFrames.jl/pull/1449). So in the meantime if you want, add a temporary `push!` method in deprecated.jl to make this work until the deprecation is removed.

Reply:
This completely fell off my radar. I'll try to take care of it now.